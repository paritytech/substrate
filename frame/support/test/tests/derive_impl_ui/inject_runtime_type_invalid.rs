use frame_support::{*, pallet_prelude::inject_runtime_type};
use static_assertions::assert_type_eq_all;

pub trait Config {
    type RuntimeInfo;
}

type RuntimeInfo = u32;

struct Pallet;

#[register_default_impl(Pallet)]
impl Config for Pallet {
    #[inject_runtime_type]
    type RuntimeInfo = ();
}

struct SomePallet;

#[derive_impl(Pallet)] // Injects type RuntimeInfo = RuntimeInfo;
impl Config for SomePallet {}

assert_type_eq_all!(<SomePallet as Config>::RuntimeInfo, u32);

fn main() {}
