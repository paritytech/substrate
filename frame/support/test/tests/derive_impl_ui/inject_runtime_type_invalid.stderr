error: `#[inject_runtime_type]` can only be attached to `RuntimeCall`, `RuntimeEvent`, `RuntimeOrigin` or `PalletInfo`
  --> tests/derive_impl_ui/inject_runtime_type_invalid.rs:15:5
   |
15 |     type RuntimeInfo = ();
   |     ^^^^^^^^^^^^^^^^^^^^^^

error[E0046]: not all trait items implemented, missing: `RuntimeInfo`
  --> tests/derive_impl_ui/inject_runtime_type_invalid.rs:13:1
   |
5  |     type RuntimeInfo;
   |     ---------------- `RuntimeInfo` from trait
...
13 | impl Config for Pallet {
   | ^^^^^^^^^^^^^^^^^^^^^^ missing `RuntimeInfo` in implementation
